Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -3013,20 +3013,35 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
 }
 
 
-void WebExtensionContext::handleContentRuleListNotificationForTab(WebExtensionTab& tab, const URL& url, WebCore::ContentRuleListResults::Result)
+bool WebExtensionContext::handleContentRuleListNotificationForTab(WebExtensionTab& tab, const URL& url, WebCore::ContentRuleListResults::Result)
 {
     incrementActionCountForTab(tab, 1);
 
     if (!hasPermission(_WKWebExtensionPermissionDeclarativeNetRequestFeedback) && !(hasPermission(_WKWebExtensionPermissionDeclarativeNetRequest) && hasPermission(_WKWebExtensionPermissionActiveTab)))
-        return;
-
-    // FIXME: rdar://118940129 - Set a timer to purge old matched rules.
+        return false;
 
     m_matchedRules.append({
         url,
         WallTime::now(),
         tab.identifier()
     });
+
+    return true;
+}
+
+bool WebExtensionContext::purgeMatchedRulesFromBefore(const WallTime& startTime)
```

Comment:
See below, but I don't think the `bool` return is needed.

Reply:
I responded to that comment. I think it is necessary due to the fact one of these is on the controller the other is on the context.